lib/utilities/sign.js

```diff
@@ -58,20 +155,85 @@ async function hardwareSignatures(hwApp, ptx, account, enc) {
  * @param {[]String}   paths  - return signature in hex
  */
 async function hardwareTransaction(hwApp, mtx, inputs, paths) {
-  const ledgerInputs = [];
-  for (const [i, input] of mtx.inputs.entries()) {
-    const li = new LedgerTXInput({
-      tx: inputs[i].toRaw(),
-      index: input.prevout.index,
-      path: paths[i],
-      witness: inputs[i].hasWitness(),
-    });
-    ledgerInputs.push(li);
-  }
+  switch (hwApp.type) {
+    case 'ledger': {
+      const ledgerInputs = [];
+      for (const [i, input] of mtx.inputs.entries()) {
+        const li = new LedgerTXInput({
+          tx: inputs[i].toRaw(),
+          index: input.prevout.index,
+          path: paths[i],
+          witness: inputs[i].hasWitness(),
+        });
+        ledgerInputs.push(li);
+      }
+
+      await hwApp.signTransaction(mtx, ledgerInputs);
+
+      return mtx;
+    }
+    case 'trezor': {
+      // TrezorConnect uses their backend to fetch
```

Comment:
Ledger has two different methods for signing and returning tx vs just fetching signature while Trezor has one method that returns both the raw tx and the signature. This causes the API to be a bit redundant in Trezor's case, as Ledger was the first client added to this codebase